week2/homework/src/methodList.js

```diff
@@ -0,0 +1,114 @@
+          console.log('no data found');
+        }
+        console.log(error);
+      } else {
+        const data = JSON.parse(todoList);
+        if (todoItem !== '.' && data[todoItem]) {
+          let i = todoItem;
+          do {
+            delete data[i];
+            if (data[++i]) data[--i] = data[++i];
+            i;
+          } while (i <= Object.keys(data).length);
+          fs.writeFile('./todoList.json', JSON.stringify(data), 'utf8', () => {
+            console.log(data);
+          });
+        } else {
+          fs.writeFile('./todoList.json', JSON.stringify({}), 'utf8', () => {
+            console.log({});
+          });
+        }
+      }
+    });
+  },
+  list: function list() {
+    fs.readFile('./todoList.json', 'utf8', (error, todoList) => {
+      if (error) {
+        if (error.code === 'ENOENT') {
+          console.log('no data found');
+        }
+        console.log(error);
+      } else {
+        const data = JSON.parse(todoList);
+        console.log(data);
+      }
+    });
+  },
+  reset: function reset() {
+    fs.readFile('./todoList.json', 'utf8', error => {
+      if (error) {
+        if (error.code === 'ENOENT') {
+          console.log('no data found');
+        }
+        console.log(error);
+      } else {
+        fs.writeFile('./todoList.json', JSON.stringify({}), 'utf8', () => {
+          console.log({});
+        });
+      }
+    });
+  },
+  update: function update(todoItem, todoValue) {
+    fs.readFile('./todoList.json', 'utf8', (error, todoList) => {
+      if (error) {
+        if (error.code === 'ENOENT') {
+          console.log('no data found');
+        }
+        console.log(error);
+      } else {
+        const data = JSON.parse(todoList);
+        data[todoItem] = todoValue;
+        fs.writeFile('./todoList.json', JSON.stringify(data), 'utf8', () => {
+          console.log(data);
+        });
+      }
+    });
+  },
+  help: function help() {
+    Object.keys(this).forEach(key => {
+      key === 'add'
+        ? console.log(`The '${key} function' helps you to ${key} new things on the todo list`)
+        : key === 'remove'
+        ? console.log(
+            `The '${key} function' helps you to ${key} things on the todo list. To ${key} all of the list please use dot(.) after the ${key} command. You can use reset function too.`,
+          )
+        : key === 'list'
+        ? console.log(`The '${key} function' helps you to ${key} all of things on the todo list`)
+        : key === 'reset'
+        ? console.log(
+            `The '${key} function', ${key}s the todo list.Everyone deserves a fresh start`,
+          )
+        : key === 'update'
+        ? console.log(
+            `The ${key} function allow you to update respected todoItem. Syntax: update "number" "new value"`,
+          )
+        : console.log(
+            `The '${key} function' is designed to ${key} you. You may use it with ${key} command or leaving the rest blank`,
+          );
+    });
+  },
```

Comment:
Help command shows:

    ➜ node . help
    The 'global function' is designed to global you. You may use it with global command or leaving the rest blank
    The 'process function' is designed to process you. You may use it with process command or leaving the rest blank
    The 'Buffer function' is designed to Buffer you. You may use it with Buffer command or leaving the rest blank
    The 'clearImmediate function' is designed to clearImmediate you. You may use it with clearImmediate command or leaving the rest blank
    The 'clearInterval function' is designed to clearInterval you. You may use it with clearInterval command or leaving the rest blank
    The 'clearTimeout function' is designed to clearTimeout you. You may use it with clearTimeout command or leaving the rest blank
    The 'setImmediate function' is designed to setImmediate you. You may use it with setImmediate command or leaving the rest blank
    The 'setInterval function' is designed to setInterval you. You may use it with setInterval command or leaving the rest blank
    The 'setTimeout function' is designed to setTimeout you. You may use it with setTimeout command or leaving the rest blank

I don't think this is what you want, I guess if you replace `this` with `module.exports`, like

    Object.keys(module.exports).forEach(key => { /* ... */ }

You will get what you expect, but still it can be written simpler just like a single text.